src/frontends/onnx/frontend/src/op/eye_like.cpp

```diff
@@ -116,32 +46,22 @@ OutputVector eye_like(const Node& node) {
                      input_rank.get_length(),
                      " is unsupported, only 2D shapes are supported");
 
-    const auto shift = node.get_attribute_value<std::int64_t>("k", 0);
-
-    std::int64_t dtype;
     element::Type target_type;
     if (node.has_attribute("dtype")) {
-        dtype = node.get_attribute_value<std::int64_t>("dtype");
+        std::int64_t dtype = node.get_attribute_value<std::int64_t>("dtype");
         target_type = common::get_ngraph_element_type(dtype);
     } else {
         target_type = input.get_element_type();
     }
 
     const auto input_shape = std::make_shared<default_opset::ShapeOf>(input);
+    const auto dims = detail::get_shape_width_and_height(input_shape);
+    const auto width = dims.at(0);
+    const auto height = dims.at(1);
+    const auto k =
+        default_opset::Constant::create(ngraph::element::i64, {1}, {node.get_attribute_value<std::int64_t>("k", 0)});
 
-    const auto component_dimensions = detail::eyelike_component_dimensions(input_shape, shift);
-    const auto identity_matrix = detail::square_identity_matrix(component_dimensions.at(0), target_type);
-
-    const auto pads_begin =
-        std::make_shared<default_opset::Concat>(OutputVector{component_dimensions.at(3), component_dimensions.at(1)},
-                                                0);
-    const auto pads_end =
-        std::make_shared<default_opset::Concat>(OutputVector{component_dimensions.at(4), component_dimensions.at(2)},
-                                                0);
-
-    const auto zero = default_opset::Constant::create(target_type, Shape{}, {0});
-    const auto output =
-        std::make_shared<default_opset::Pad>(identity_matrix, pads_begin, pads_end, zero, ov::op::PadMode::CONSTANT);
+    const auto output = std::make_shared<ov::op::v9::Eye>(height, width, k, target_type);
```

Comment:
```suggestion
    const auto output = std::make_shared<default_opset::Eye>(height, width, k, target_type);
```

Reply:
It has to wait when default `opset` will changed to version 9, now is version 8 and which not support the `eye`.